### PR TITLE
Update documentation for DataFrame methods

### PR DESCRIPTION
**Pull request by devin-petersohn**
## Related issue number

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] tests added and passing

### CODE REVIEW

File: docs/dataframe_supported.rst [hunk 1]
```diff
@@ -0,0 +1,458 @@
+pandas.DataFrame supported APIs
+===============================
+
+The following table lists both implemented and not implemented methods. If you have need
+of an operation that is listed as not implemented, feel free to open an issue on the
+`GitHub repository`_. Contributions are also welcome!
+
++---------------------------+---------------------------------+----------------------------------------------------+
+| DataFrame method          | Modin Implementation? (Y/N/P/D) | Notes for Current implementation                   |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: Let's define `Y/N/P/D` for clairty

File: docs/dataframe_supported.rst [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+| ``add_prefix``            | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``add_suffix``            | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``agg``                   | Y                               | - Passing a dictionary for the ``func`` parameter  |
+|                           |                                 | defaults to pandas                                 |
+|                           |                                 | - Passing the string name of a numpy operation for |
+| ``aggregate``             |                                 | the ``func`` parameter defaults to pandas          |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``align``                 | D                               | Defaults to pandas                                 |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: May not need to explicitly mention `Defaults to pandas` in the notes if `D` represents defaults to pandas

File: docs/dataframe_supported.rst [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``combine_first``         | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``compound``              | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``consolidate``           | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``convert_objects``       | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``copy``                  | Y                               |                                                    |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: Do we do deep and shallow copies properly?

File: docs/dataframe_supported.rst [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``mode``                  | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``mul``                   | Y                               | See ``add``                                        |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``multiply``              | Y                               | See ``add``                                        |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``ndim``                  | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``ne``                    | Y                               | See ``__ne__``                                     |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: `__ne__` isnt in the documentation

File: docs/dataframe_supported.rst [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``pct_change``            | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``pipe``                  | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``pivot``                 | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``pivot_table``           | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``plot``                  | Y                               |                                                    |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: Not implemented

File: docs/dataframe_supported.rst [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``pow``                   | Y                               | See ``add``                                        |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``prod``                  | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``product``               | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``quantile``              | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``query``                 | Y                               | Local variables not yet supported                  |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: Perhaps partial instead of Y since local variables are not supported?

File: docs/dataframe_supported.rst [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``swapaxes``              | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``swaplevel``             | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``tail``                  | Y                               |                                                    |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``take``                  | D                               | Defaults to pandas                                 |
++---------------------------+---------------------------------+----------------------------------------------------+
+| ``to_clipboard``          | Y                               |                                                    |
```
Comment (williamma12) [inline docs/dataframe_supported.rst]: Most of the `to_` functions are not implemented

File: docs/dataframe_supported.rst (hunk 3 above)
Comment (devin-petersohn) [reply docs/dataframe_supported.rst]: Yes, we do now as of 0.5.1

**Pull request merged**